wagtail/snippets/views/snippets.py

```diff
@@ -177,18 +178,40 @@ def get_columns(self):
             *super().get_columns(),
         ]
 
-    def get_list_dropdown_buttons(self, instance, parent_context):
-        buttons = super().get_list_dropdown_buttons(instance, parent_context)
+    def get_list_buttons(self, instance, parent_context):
+        more_buttons = self.get_list_dropdown_buttons(instance, parent_context)
         next_url = self.request.path
         button_hooks = hooks.get_hooks("register_snippet_listing_buttons")
+        list_buttons = []
 
         for hook in button_hooks:
-            buttons.extend(hook(instance, self.request.user, next_url))
+            hook_buttons = hook(instance, self.request.user, next_url)
+            for button in hook_buttons:
+                if isinstance(button, BaseDropdownMenuButton):
+                    # If the button is a dropdown menu, add it to the top-level
+                    # because we do not support nested dropdowns
+                    list_buttons.append(button)
+                else:
+                    # Otherwise, add it to the default "More" dropdown
+                    more_buttons.append(button)
+
+        list_buttons.append(
+            ButtonWithDropdown(
+                buttons=more_buttons,
+                icon_name="dots-horizontal",
+                attrs={
+                    "aria-label": _("More options for '%(title)s'")
+                    % {"title": str(instance)},
+                },
+            )
+        )
 
+        # Pass the top-level buttons to the hooks, so top-level non-dropdown
+        # buttons can still be added.
         for hook in hooks.get_hooks("construct_snippet_listing_buttons"):
-            hook(buttons, instance, self.request.user, parent_context)
+            hook(list_buttons, instance, self.request.user, parent_context)
```

Comment:
It looks like the most common use of the `construct_snippet_listing_buttons` hook is to remove wagtailadmin's default actions from the list - see for example https://github.com/nationalarchives/ds-wagtail/blob/8d555934719b06abaae6ff82de12571447c9aade/etna/feedback/wagtail_hooks.py#L8 - so I think this really has to act on the same list as the 'register' hook, namely `more_buttons` - otherwise we're removing the ability to do that.